### ISSUE TITLE
Make Green Island independent from QtCompositor

### ISSUE THREAD

**Issue by plfiorini** [greenisland/greenisland#104]
Making changes to QtCompositor and then adapting Green Island to it is hard, it often happens that we are out of sync and then require a special branch or some patches (like Wayland 1.6 support which cannot be applied upstream because Qt CI has a lower libwayland version).

The QtCompositor API is not stable, needs an API review and a few things are missing before an official release is possible, there's no sign of interest in doing this in a reasonable (for us) time frame. There are rumors of a QtCompositor overhaul from Qt CS 2015 but ETA is unknown and nobody took care of writing notes from the Wayland session (unfortunately I didn't make it to Qt CS 2015 due to real work deadlines).

TL;DR: Make a QtCompositor copy here and then merge stuff back upstream when we are ready.   It will also make it easier for distros to ship our stuff (no dependency with custom branches or patchsets).

Hopefully we'll come up with things that are useful like not exposing the private implementation, better global interface and surface role handling, etc... and share the result with QtWayland.

This means that from 0.7.0 we'll drop the QtWayland dependency at least until the QtCompositor API is set in stone and released.

When that happens well use the QtCompositor library again.
There will still be a need for Green Island because it will always offer the non general purpose stuff like additional protocols, server-side decoration, a more complete client window interface, effects, plugins etc...
